### PR TITLE
Gridcoin update to v3.4.2.5

### PR DESCRIPTION
**Pull request by ghost**
Added latest commit. Is there a way to make this process automatic Martin?

~William

### CODE REVIEW

**Comment by Git-Jiro** [comment]
If we omit the ", revsion => '...'" from the formula, it will always pull the latest version from the repository.

But I do not like this, because:
In my humble opinion it is better to have a little control about what gets pulled from the repo.
If some commit to the Gridcoin-Research repo would break the OS X build, it would also break the brew formula with it. But only if we always pull the latest version.

If we pull the latest known good version, this will not happen.

Basically what I want to avoid is this:
Somebody else uses this brew formula. --> It does not work immediately. --> Bad user experience.

**Comment by ghost** [comment]
I was wondering about removing the revision part, but was assuming it would pick up the latest good commit like the one we took today.  You make a good point Martin. And I agree, it helps to have a little control.

Check me out at acey1. I've forked from you and gridcoin in order to reverse engineer what you all have done, in order to facilitate my learning. Today I brewed my acey1 clone of your formula in order to compile my own dmg without waiting for you to merge from my pull request.  It worked! So now I can literally test my pull before sending you the request. This is cool stuff!

Thanks for introducing me to this place Martin! If it were not for you, I would not be running grc on osx right now.  

So a question please: What is the feasibility of repairing the upgrade, download blocks, and restart functions in gridcoin itself? Everyone is having trouble with this, because we are using the old Blackcoin code complete with icons, bad IPs, and no telling what else.

Best regards,
William

**Comment by Git-Jiro** [comment]
Hi William,
Glad I could introduce you to the wonderful world of software development! :-)

Regarding the upgrade feature:
Disclosure: I have never written software with an upgrade feature myself so i can only make educated guesses. But here they are:

For this feature to work it is necessary to have a designated location on the web were the software can check for new upgrades and download them. Since there was not even a DMG for OS X a few days ago, I also assume there is no designated location for OS X updates right now. This feature will not work if there is no automation to publish new DMGs somewhere. But first some has to decide where "somewhere" is and what is considered a stable version.

Regarding the icon:
There is a nice Gridcoin symbol in the splash screen of the app.
With a litte bit of Gimp or Photoshop it should be doable to extract this into an icon.
I assume this is just not a priority right now for the main developers.

Regarding download blocks:
This functionality works in the debug console, if i remember correctly.

Best regards,
Martin

**Comment by ghost** [comment]
Discussion please: "For this feature to work it is necessary to have a designated location on the web were the software can check for new upgrades and download them."

Well...
Isn't this already being done with Git? Is it possible (for osx anyway) & (I have no clue how the windows guys do it)  to develop a pull request to gridcoin showing an upgrade link to the grc repo? For now the ssh for the latest version could be manually updated by pull request from us. I am assuming that when the client upgrades, it only replaces the gridcoin version part of the app itself. So in other words, could an upgrade button import the version changes from Git into an existing app like I describe?

The compile guys at unix told me their upgrade was broken also, but was never code updated so still tries the old Blackcoin IP. 

I might could track down the icon and replace it. Test it, and pull request. That would be a fun learning experience for me. Which directory would it lie in?  Ha Ha!
